c_src/membrane_agora_plugin/sink.cpp

```diff
@@ -151,12 +151,18 @@ UNIFEX_TERM update_audio_stream_format(UnifexEnv *env, int sampleRate,
 }
 
 UNIFEX_TERM write_audio_data(UnifexEnv *env, UnifexPayload *payload,
-                             SinkState *state) {
+                             CodecAudio codec, SinkState *state) {
   agora::rtc::EncodedAudioFrameInfo audioFrameInfo;
   audioFrameInfo.sampleRateHz = state->sampleRate;
   audioFrameInfo.numberOfChannels = state->numberOfChannels;
   audioFrameInfo.samplesPerChannel = state->samplesPerChannelPerFrame;
-  audioFrameInfo.codec = agora::rtc::AUDIO_CODEC_TYPE::AUDIO_CODEC_AACLC;
+
+  if (codec == CODEC_AUDIO_AAC) {
+    audioFrameInfo.codec = agora::rtc::AUDIO_CODEC_TYPE::AUDIO_CODEC_AACLC;
+  } else if (codec == CODEC_AUDIO_OPUS) {
+    audioFrameInfo.codec = agora::rtc::AUDIO_CODEC_TYPE::AUDIO_CODEC_OPUS;
+  }
```

Comment:
[NIT] I would warn if other codec was passed here. I know that now the only options for CodecAudio are `CODEC_AUDIO_AAC` and `CODEC_AUDIO_OPUS`, but it might cause trouble in the future if we were to add some further codecs.